tools/ci_build/github/linux/docker/scripts/training/azure_scale_set_vm_mount_test_data.sh

```diff
@@ -1,30 +1,70 @@
 #! /usr/bin/env bash
-if [ -d "/bert_data" ]; then
-sudo umount /bert_data
-fi
-if [ -d "/bert_data" ]; then
-sudo rm -rf /bert_data
-fi
 
-sudo mkdir /bert_data
+function credentialize () {
+    if [ ! -d "/etc/smbcredentials" ]; then
+    sudo mkdir /etc/smbcredentials
+    fi
 
-if [ ! -d "/etc/smbcredentials" ]; then
-sudo mkdir /etc/smbcredentials
-fi
+    if [ -f "/etc/smbcredentials/orttrainingtestdata.cred" ]; then
+    sudo rm /etc/smbcredentials/orttrainingtestdata.cred
+    fi
 
-if [ -f "/etc/smbcredentials/orttrainingtestdata.cred" ]; then
-sudo rm /etc/smbcredentials/orttrainingtestdata.cred
-fi
+    # to create orttrainingtestdata.cred, I have to do: 'sudo bash -c ...'
+    sudo bash -c 'echo "username=orttrainingtestdata" >> /etc/smbcredentials/orttrainingtestdata.cred'
 
-# to create orttrainingtestdata.cred, I have to do: 'sudo bash -c ...'
-sudo bash -c 'echo "username=orttrainingtestdata" >> /etc/smbcredentials/orttrainingtestdata.cred'
+    # $1 get removed (do defend injection attack?) if I do 'sudo bash -c...'
+    # to enable 'sudo echo...' I need to 'sudo chmod 777...' first.
+    sudo chmod 777 /etc/smbcredentials/orttrainingtestdata.cred
+    sudo echo "password=$1" >> /etc/smbcredentials/orttrainingtestdata.cred
```

Comment:
I know this is pre-existing code, but I can't help saying that doing `chmod 777` for credentials is terribly unsafe. If the VM is compromised, this file is out in the open

Maybe we should create a work item to fix this soon

Reply:
I agree that this is not safe, but I think if the VM is compromised, we may have other problems as well.
And the storage account access key can be reset if this situation were to arise.
But I agree, we can investigate means to not need this.
@liqunfu why does the credentials need ```777``` access?

Reply:
So it will not block the next run. I suppose.